gitolith-core/src/changelog.rs

```diff
@@ -0,0 +1,95 @@
+use crate::error::Result;
+use crate::release::Release;
+use std::collections::HashMap;
+use tera::{
+	Context as TeraContext,
+	Result as TeraResult,
+	Tera,
+	Value,
+};
+
+/// Changelog generator.
+#[derive(Debug)]
+pub struct Changelog {
+	tera: Tera,
+}
+
+impl Changelog {
+	/// Constructs a new instance.
+	pub fn new(template: String) -> Result<Self> {
+		let mut tera = Tera::default();
+		tera.add_raw_template("changelog_template", &template)?;
+		tera.register_filter("capitalize_first", Self::capitalize_first_filter);
+		Ok(Self { tera })
+	}
+
+	/// Filter for capitalizing the first character of a string.
+	fn capitalize_first_filter(
+		value: &Value,
+		_: &HashMap<String, Value>,
+	) -> TeraResult<Value> {
+		let mut s =
+			tera::try_get_value!("capitalize_first_filter", "value", String, value);
+		let mut c = s.chars();
+		s = match c.next() {
+			None => String::new(),
+			Some(f) => f.to_uppercase().collect::<String>() + c.as_str(),
+		};
+		Ok(tera::to_value(&s).unwrap())
```

Comment:
I don't like the unwrap here. Do we have a logical explanation of why we have this in here?

Reply:
No, I just forgot.